x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/create_field/create_field.tsx

```diff
@@ -76,6 +98,24 @@ export const CreateField = React.memo(function CreateFieldComponent({
     }
   };
 
+  const [referenceFieldComboValue, setReferenceFieldComboValue] = useState<any>();
+  const [nameValue, setNameValue] = useState<any>();
+  const [inferenceIdComboValue, setInferenceIdComboValue] = useState<any>();
+  const [semanticFieldType, setSemanticTextFieldType] = useState<any>();
+
+  useFieldEffect(form, 'referenceField', setReferenceFieldComboValue);
+  useFieldEffect(form, 'inferenceId', setInferenceIdComboValue);
+  useFieldEffect(form, 'name', setNameValue);
+
+  const fieldTypeValue = form.getFields()?.type?.value;
+  useEffect(() => {
+    const type = form.getFields()?.type?.value as any;
```

Comment:
We should (almost) never be using `as any` in code outside of tests. `as any` completely overrides the type system. Define an actual type and use it, then use the typing system to your advantage.

Reply:
Updated